### PR TITLE
feat!: Large rewrite with testing, hover and partial completions

### PR DESCRIPTION
**Pull request by neubaner**
This PR adds a lot of legwork to make the Razor LSP communicate with Roslyn correctly, hover support and partial completions for the C# portions of Razor file.

### CODE REVIEW

**Review by tris203 — CHANGES_REQUESTED** [review]
This looks awesome, thank you so much

File: lua/rzls/handlers/init.lua [hunk 1]
```diff
@@ -137,5 +160,11 @@ return {
     ["razor/csharpPullDiagnostics"] = not_implemented,
     ["textDocument/colorPresentation"] = not_supported,
 
-    ["razor/initialize"] = not_implemented,
```
[the remark this answers is not in the document]
Comment (gneubaner) [reply lua/rzls/handlers/init.lua:140]: "razor/initialize" is actually a handler of the Roslyn LSP.

[the remark this answers is not in the document]
Comment (neubaner) [reply lua/rzls/handlers/init.lua:140]: "razor/initialize" is a handler for the Roslyn LSP.

File: lua/rzls/razor.lua [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        vim.api.nvim_buf_add_highlight(
+            buf,
+            -1,
+            "QuickFixLine",
+            response.position.line,
+            response.position.character,
+            response.position.character + 3
+        )
+    end)
+end, {})
```
[the remark this answers is not in the document]
Comment (neubaner) [reply lua/rzls/razor.lua]: It was leftover code, already got rid of it :)

**Pull request merged**